multimodal/src/autogluon/multimodal/models/sam.py

```diff
@@ -116,15 +371,43 @@ def forward(
         -------
             A dictionary with mask predictions.
         """
-        rets = self.model(batch[self.image_key], multimask_output=False)
-        pred_masks = rets.pred_masks[:, 0, :, :, :]
-        pred_masks = F.interpolate(
-            pred_masks, (self.image_size, self.image_size), mode="bilinear", align_corners=False
-        )
-        if self.training:
-            return {self.prefix: {LOGITS: pred_masks}}
+        # binary
+        if self.num_classes == 1:
+            rets = self.model(batch[self.image_key], multimask_output=False)
+            pred_masks = rets.pred_masks[:, 0, :, :, :]
+            pred_masks = F.interpolate(
+                pred_masks, (self.image_size, self.image_size), mode="bilinear", align_corners=False
+            )
+            if self.training:
+                return {self.prefix: {LOGITS: pred_masks}}
+            else:
+                return {self.prefix: {LOGITS: torch.sigmoid(pred_masks), LABEL: batch[self.label_key]}}
+
+        # multi-class
         else:
-            return {self.prefix: {LOGITS: pred_masks, LABEL: batch[self.label_key]}}
+            rets, class_predictions = self.model(batch[self.image_key], multimask_output=False)
+            pred_masks = rets.pred_masks[:, 0, :, :, :]
+            pred_classes = class_predictions[:, 0, :, :]
+            pred_masks = F.interpolate(
+                pred_masks, (self.image_size, self.image_size), mode="bilinear", align_corners=False
+            )
+            if self.training:
+                return {self.prefix: {LOGITS: pred_masks, CLASS_LOGITS: pred_classes}}
+            else:
+                processed_results = []
+                for mask_cls_result, mask_pred_result in zip(pred_classes, pred_masks):
+                    r = self.semantic_inference(mask_cls_result, mask_pred_result)
+                    processed_results.append(r.unsqueeze(0))
+                processed_results = torch.cat(processed_results, dim=0)
```

Comment:
What's the shape of concatenated tensor? I guess we need the shape `(bs, num_classes, height, width)`?

Reply:
if `r` has shape `(num_classes, height, width)`, shouldn't we use `torch.stack`?